src/Microsoft.OData.Client/ALinq/ExpressionWriter.cs

```diff
@@ -372,6 +373,22 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
                         this.builder.Append(UriHelper.FORWARDSLASH).Append(UriHelper.DOLLARSIGN).Append(UriHelper.COUNT);
                         return m;
                     }
+                    else if (sequenceMethod == SequenceMethod.Contains)
+                    {
+                        // First argument is the collection expression
+                        // Second argument is the value expression
+                        // Note that arguments must be reordered for the IN operator
+                        // e.g. ctx.CreateQuery<Product>("Products").Where(p => (new [] { "Milk", "Cheese", "Donut"}).Contains(p.Name))

```

Comment:
>(p => (new [] { "Milk", "Cheese", "Donut"}).Contains(p.Name)) [](start = 74, length = 61)

can we do:

Where(p => p.CollectionProperties.Contains('abc'))?

Reply:
See my suggested added test case...

---
In reply to: [539597874](https://github.com/OData/odata.net/pull/1868#discussion_r539597874) [](ancestors = 539597874)

Reply:
I added the suggested test cases but they simply reflect the old behavior, because I never meant to add support for these type of expressions.

Reply:
fair enough -- we'll create a work item to add the other form.

---
In reply to: [544603939](https://github.com/OData/odata.net/pull/1868#discussion_r544603939) [](ancestors = 544603939)